smartsim/launcher/step/lsfStep.py

```diff
@@ -0,0 +1,236 @@
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import os
+from itertools import product
+
+from ...error import SSConfigError
+from ...utils import get_logger
+from .step import Step
+
+logger = get_logger(__name__)
+
+
+class BsubBatchStep(Step):
+    def __init__(self, name, cwd, batch_settings):
+        """Initialize a LSF bsub step
+
+        :param name: name of the entity to launch
+        :type name: str
+        :param cwd: path to launch dir
+        :type cwd: str
+        :param batch_settings: batch settings for entity
+        :type batch_settings: BatchSettings
+        """
+        super().__init__(name, cwd)
+        self.batch_settings = batch_settings
+        self.step_cmds = []
+        self.managed = True
+
+    def get_launch_cmd(self):
+        """Get the launch command for the batch
+
+        :return: launch command for the batch
+        :rtype: list[str]
+        """
+        script = self._write_script()
+        return [self.batch_settings.batch_cmd, script]
+
+    def add_to_batch(self, step):
+        """Add a job step to this batch
+
+        :param step: a job step instance e.g. SrunStep
+        :type step: Step
+        """
+        launch_cmd = step.get_launch_cmd()
+        self.step_cmds.append(launch_cmd)
+        logger.debug(f"Added step command to batch for {step.name}")
+
+    def _write_script(self):
+        """Write the batch script
+
+        :return: batch script path after writing
+        :rtype: str
+        """
+        batch_script = self.get_step_file(ending=".sh")
+        output, error = self.get_output_files()
+
+        self._format_alloc_flags()
+
+        with open(batch_script, "w") as f:
+            f.write("#!/bin/bash\n\n")
+            if self.batch_settings.walltime:
+                f.write(f"#BSUB -W {self.batch_settings.walltime}\n")
+            if self.batch_settings.project:
+                f.write(f"#BSUB -P {self.batch_settings.project}\n")
+            f.write(f"#BSUB -o {output}\n")
+            f.write(f"#BSUB -e {error}\n")
+            f.write(f"#BSUB -J {self.name}\n")
+
+            # add additional bsub options
+            for opt in self.batch_settings.format_batch_args():
+                f.write(f"#BSUB {opt}\n")
+
+            for i, cmd in enumerate(self.step_cmds):
+                f.write("\n")
+                f.write(f"{' '.join((cmd))} &\n")
+                if i == len(self.step_cmds) - 1:
+                    f.write("\n")
+                    f.write("wait\n")
+        return batch_script
+
+
+class JsrunStep(Step):
+    def __init__(self, name, cwd, run_settings):
+        """Initialize a LSF jsrun job step
+
+        :param name: name of the entity to be launched
+        :type name: str
+        :param cwd: path to launch dir
+        :type cwd: str
+        :param run_settings: run settings for entity
+        :type run_settings: RunSettings
+        """
+        super().__init__(name, cwd)
```

Comment:
OK, this is a bit harder than on Slurm. What happens when a job is launched is that you can use `jslist` to see all the steps you launched. But you have to keep track of the ids (I think they start from 0 and are increased by 1) manually -> there is no step name, no other way of reconnecting your step with the ID. I guess we could add this info to the launcher, but I'm not super-confident this will work (and maybe it is beyond the scope of this PR?). Let me know.

Reply:
Lets put in a follow up ticket about this. I have a feeling that the Job manager refactor will address some of this.